File: Web_DienThoai.Server/Models/Product.cs
namespace Web_NoiThat.Server.Models;

/// <summary>
/// Schema dữ liệu Sản phẩm nội thất - bao gồm kích thước, chất liệu, giá.
/// </summary>
public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    // --- Giá ---
    public decimal Price { get; set; }
    public decimal? SalePrice { get; set; }       // Giá flash sale
    public bool IsOnSale { get; set; } = false;

    // --- Kích thước (cm) ---
    public double? LengthCm { get; set; }          // Dài
    public double? WidthCm { get; set; }   // Rộng
    public double? HeightCm { get; set; } // Cao
    public double? WeightKg { get; set; }          // Khối lượng (kg)

    // --- Phân loại ---
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public string Material { get; set; } = string.Empty;   // Gỗ, Da, Vải...
    public string Style { get; set; } = string.Empty;  // Minimalist, Indochine...
    public string Color { get; set; } = string.Empty;

    // --- Tồn kho ---
    public int Stock { get; set; } = 0;
    public bool IsActive { get; set; } = true;
    public int SoldCount { get; set; } = 0;

    // --- Ảnh ---
    public string MainImageUrl { get; set; } = string.Empty;
    public ICollection<ProductImage> SubImages { get; set; } = [];

 // --- Quan hệ ---
public ICollection<OrderItem> OrderItems { get; set; } = [];
    public ICollection<Review> Reviews { get; set; } = [];
    public ICollection<WishlistItem> WishlistItems { get; set; } = [];
    public ICollection<ProductBundle> Bundles { get; set; } = [];  // "Mua kèm"

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: Web_DienThoai.client/src/types/product.ts
// ===== PRODUCT TYPES =====

export interface ProductImage {
  id: number;
  url: string;
  altText?: string;
  sortOrder: number;
}

export interface Category {
  id: number;
  name: string;
  slug: string;
  description?: string;
  imageUrl?: string;
  icon?: string;
  sortOrder: number;
  productCount: number;
}

// 👉 Dùng cho card
export interface ProductCard {
  id: number;
  name: string;
  slug: string;

  // ⚠️ đổi sang images để khớp db.json
  images: string[];

  price: number;
  salePrice?: number;
  isOnSale: boolean;

  material: string;
  style: string;
  color: string;

  soldCount: number;
  averageRating?: number;
  reviewCount: number;
}

// 👉 Trang chi tiết
export interface ProductDetail extends ProductCard {
  description: string;
  stock: number;

  subImages?: ProductImage[];

  lengthCm?: number;
  widthCm?: number;
  heightCm?: number;
  weightKg?: number;

  category?: Category;
  bundledProducts?: ProductCard[];
  latestReviews?: Review[];
}

export interface Review {
  id: number;
  userName: string;
  userAvatarUrl?: string;
  rating: number;
  comment: string;
  imageUrls: string[];
  adminReply?: string;
  createdAt: string;
}

export interface ProductFilterParams {
  categoryId?: number;
  categorySlug?: string;
  q?: string;
  material?: string;
  style?: string;
  color?: string;
  minPrice?: number;
  maxPrice?: number;
  sortBy?: "newest" | "price_asc" | "price_desc" | "best_seller";
  page?: number;
  pageSize?: number;
}

// 👉 QUAN TRỌNG: khớp với API đã fix
export interface PaginatedProducts {
  items: ProductCard[];
  total: number;
  page: number;
  pageSize: number;
}
